### DISCUSSION TITLE
changing electrode causes error when running experiments

### DISCUSSION THREAD

**Discussion by Brushnie33** [pybamm-team/PyBaMM#2302] [Q&A]
Hello,
While I was running the lithium-plating model from the example, the codes were executed successfully. And when I changed the negative electrode to graphite_Ecker2015 along with Initial concentration in negative electrode to 12630.8, and ran the exact same experiment, I received errors. 

**The following are the codes that I have modified:**

chemistry = pybamm.parameter_sets.OKane2022
chemistry["negative electrode"] = "graphite_Ecker2015"
param_update = pybamm.ParameterValues(chemistry = chemistry)
param_update.update({"Initial concentration in negative electrode [mol.m-3]": 12630.8})

**And this is the error that I've received:**

At t = 0.0209104 and h = 4.34678e-014, the corrector convergence failed repeatedly or with |h| = hmin.
At t = 0.0209089 and h = 1.34603e-013, the corrector convergence failed repeatedly or with |h| = hmin.
At t = 0.00705044 and h = 5.80127e-014, the corrector convergence failed repeatedly or with |h| = hmin.
The linesearch algorithm failed with too small a step.
The linesearch algorithm failed with too small a step.
The linesearch algorithm failed with too small a step.
2022-09-14 14:46:42.833 - [ERROR] callbacks.on_experiment_error(226): Simulation error: .../casadi/interfaces/sundials/idas_interface.cpp:591: IDACalcIC returned "IDA_LINESEARCH_FAIL". Consult IDAS documentation.
Output exceeds the [size limit](command:workbench.action.openSettings?[). Open the full output data [in a text editor](command:workbench.action.openLargeOutput?fa6c080a-d6b7-4881-b1b8-36d83c45fe82)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File c:\apps\opentoolsuite\runtime\dos\python\3.9.7\lib\site-packages\pybamm\solvers\casadi_solver.py:690, in CasadiSolver._run_integrator(self, model, y0, inputs_dict, inputs, t_eval, use_grid, extract_sensitivities_in_solution)
    689 pybamm.logger.debug("Calling casadi integrator")
--> 690 casadi_sol = integrator(
    691     x0=y0_diff, z0=y0_alg, p=inputs_with_tmin, **self.extra_options_call
    692 )
    693 pybamm.logger.debug("Finished casadi integrator")

File c:\apps\opentoolsuite\runtime\dos\python\3.9.7\lib\site-packages\casadi\casadi.py:13453, in Function.__call__(self, *args, **kwargs)
  13451   else:
  13452 # Named inputs -> return dictionary
> 13453     return self.call(kwargs)

File c:\apps\opentoolsuite\runtime\dos\python\3.9.7\lib\site-packages\casadi\casadi.py:12324, in Function.call(self, *args)
  12257 """
  12258   Generate a Jacobian function of output oind with respect to input iind.
  12259 
   (...)
  12322 
  12323 """
> 12324 return _casadi.Function_call(self, *args)

RuntimeError: .../casadi/interfaces/sundials/idas_interface.cpp:591: IDACalcIC returned "IDA_LINESEARCH_FAIL". Consult IDAS documentation.
...
    741 except RuntimeError as e:
    742     # If it doesn't work raise error
--> 743     raise pybamm.SolverError(e.args[0])

SolverError: .../casadi/interfaces/sundials/idas_interface.cpp:591: IDACalcIC returned "IDA_LINESEARCH_FAIL". Consult IDAS documentation.

**Comment by valentinsulzer** [comment]
You need to be careful when mixing parameter sets and changing initial conditions. At least run https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/models/electrode-state-of-health.ipynb to make sure the resulting capacity is what you expect

**Reply by Brushnie33** [reply]
@tinosulzer Thanks for your reply! Do you have any suggestions on how to run the experiments if I would like to investigate the effect of cathodes materials on cells' performance (only change cathodes, but keep anodes and electrolytes/separator the same)?

**Reply by Brushnie33** [reply]
@tinosulzer Hello, I was using the example to calculate the electrode state of health for different built-in cells, and I've received errors for Ai2020 and Okane2022. I didn't try all the cells, though.
